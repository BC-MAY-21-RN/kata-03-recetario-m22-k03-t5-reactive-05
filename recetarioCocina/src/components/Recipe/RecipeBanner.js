import React, {useState} from 'react';
import {View, Image, StyleSheet, TouchableOpacity} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import IonIcon from 'react-native-vector-icons/Ionicons';
import {useNavigation} from '@react-navigation/native';

export const RecipeBanner = ({image}) => {
  const [like, setLike] = useState(false);
  const [shared, setShared] = useState(false);
  const onLikePress = () => setLike(!like);
  const onSharedPress = () => setShared(!shared);
  const navigation = useNavigation();
  return (
    <View style={styles.bannerContainer}>
      <TouchableOpacity style={styles.iconCloseStyle}>
        <IonIcon
          name="close"
          size={30}
          style={styles.iconClosedStyle}
          onPress={() => {
            navigation.goBack();
          }}
        />
      </TouchableOpacity>
      <TouchableOpacity style={styles.iconLikeStyle}>
        <Icon
          name="heart-o"
          size={30}
          style={like ? styles.iconLikedStyle : styles.iconUnlikedStyle}
          onPress={onLikePress}
        />
      </TouchableOpacity>
      <TouchableOpacity style={styles.iconShareStyle} onPress={onSharedPress}>
        <IonIcon
          name="share-outline"
          style={shared ? styles.iconSharedStyle : styles.iconUnsharedStyle}
          size={30}
        />
      </TouchableOpacity>
      <View style={styles.bannerShadow} />
      <Image
        source={{
          uri: image,
        }}
        style={styles.recipeImage}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  iconLikeStyle: {
    position: 'absolute',
    right: 10,
    zIndex: 50,
    top: 15,
  },
  iconUnlikedStyle: {
    position: 'absolute',
    right: 15,
    zIndex: 50,
    top: 15,
    color: 'white',
  },
  iconLikedStyle: {
    position: 'absolute',
    right: 15,
    zIndex: 50,
    top: 15,
    color: '#FF479C',
  },
  iconUnsharedStyle: {
    position: 'absolute',
    right: 15,
    zIndex: 50,
    top: 15,
    color: 'white',
  },
  iconShareStyle: {
    position: 'absolute',
    right: 50,
    zIndex: 50,
    top: 15,
  },
  iconSharedStyle: {
    position: 'absolute',
    right: 15,
    zIndex: 50,
    top: 15,
    color: '#FF479C',
  },
  iconCloseStyle: {
    position: 'absolute',
    left: 60,
    zIndex: 50,
    top: 15,
  },
  iconClosedStyle: {
    position: 'absolute',
    right: 15,
    zIndex: 50,
    top: 15,
    color: 'white',
  },
  recipeImage: {
    width: '100%',
    height: 300,
  },
  bannerContainer: {
    position: 'relative',
  },
  bannerShadow: {
    backgroundColor: 'rgba(0,0,0,.4)',
    position: 'absolute',
    zIndex: 10,
    height: '100%',
    width: '100%',
  },
});
