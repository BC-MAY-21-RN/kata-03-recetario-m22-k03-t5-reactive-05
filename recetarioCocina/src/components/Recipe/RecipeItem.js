import React from 'react';
import {View, Image, StyleSheet, Text, TouchableOpacity} from 'react-native';
export const RecipeItem = ({recipe, trending}) => {
  return (
    <TouchableOpacity style={styles.recipeCardStyle} key={recipe.id}>
      <View>
        <Image
          source={{
            uri: recipe.img,
          }}
          style={[
            styles.recipeImageTrending,
            trending ? null : styles.imageSizeRecent,
          ]}
        />
        <Text style={styles.subtitle}>{recipe.title}</Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  title: {
    fontSize: 23,
    color: '#FF479C',
    marginLeft: 30,
    marginTop: 20,
    marginBottom: 15,
  },
  subtitle: {
    color: 'white',
    marginTop: 10,
  },
  recipeCardStyle: {
    marginRight: 10,
  },
  recipeImageTrending: {
    width: 150,
    height: 150,
    borderRadius: 15,
  },
  imageSizeRecent: {
    width: 200,
    height: 200,
  },
});
