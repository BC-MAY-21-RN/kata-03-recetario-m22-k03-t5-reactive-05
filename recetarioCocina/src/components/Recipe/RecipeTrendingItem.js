import React from 'react';
import {View, Image, StyleSheet, Text, TouchableOpacity} from 'react-native';

export const RecipeTrendingItem = ({recipe}) => {
  return (
    <TouchableOpacity style={styles.recipeCardStyle} key={recipe.id}>
      <View>
        <Image
          source={{
            uri: recipe.img,
          }}
          style={styles.recipeImageTrending}
        />
        <Text style={styles.subtitle}>{recipe.title}</Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  title: {
    fontSize: 23,
    color: '#FF479C',
    marginLeft: 30,
    marginTop: 20,
    marginBottom: 15,
  },
  subtitle: {
    color: 'white',
    marginTop: 10,
  },
  recipeCardStyle: {
    marginRight: 10,
  },
  recipeImageTrending: {
    width: 200,
    height: 200,
    borderRadius: 15,
  },
});
