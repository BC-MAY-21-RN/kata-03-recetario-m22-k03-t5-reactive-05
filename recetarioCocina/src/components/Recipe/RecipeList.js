import React from 'react';
import {ScrollView, StyleSheet} from 'react-native';
import {RecipeItem} from './RecipeItem';

const RecipeList = ({recipes, trending}) => {
  return (
    <ScrollView horizontal style={styles.scrollView}>
      {recipes.map(recipe => (
        <RecipeItem recipe={recipe} key={recipe.id} trending={trending} />
      ))}
    </ScrollView>
  );
};

export default RecipeList;

const styles = StyleSheet.create({
  scrollView: {
    width: '90%',
    padding: 10,
    marginRight: '5%',
    marginLeft: '5%',
  },
});
