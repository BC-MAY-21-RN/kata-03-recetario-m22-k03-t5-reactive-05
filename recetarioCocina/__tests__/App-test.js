import 'react-native';
import {RecipeItem} from '../src/components/Recipe/RecipeItem';
import {RecipeList} from '../src/components/Recipe/RecipeList';
import React from 'react';
import renderer from 'react-test-renderer';
import recipes from '../db/data.json';

test('Demostrate the array is not empty', () => {
  expect(RecipeItem).not.toBeNull();
});
describe('Recipe', () => {
  it('RecipeItem component render', () => {
    const RenderedItem = renderer
      .create(<RecipeItem recipe={recipes[0]} category="Trending" trending />)
      .toJSON();
    expect(RenderedItem).toMatchSnapshot();
  });
  it('RecipeList Component Render', () => {
    const RenderedRecipeListItem = renderer
      .create(<RecipeList recipes={recipes} trending category="Trending" />)
      .toJSON();
    expect(RenderedRecipeListItem).toMatchSnapshot();
  });
});
