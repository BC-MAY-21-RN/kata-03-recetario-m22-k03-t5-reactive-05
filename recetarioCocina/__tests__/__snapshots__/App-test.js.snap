// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe RecipeItem component render 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginRight": 10,
      "opacity": 1,
    }
  }
>
  <View>
    <Image
      source={
        Object {
          "uri": "https://t2.rg.ltmcdn.com/es/posts/2/7/7/lemon_pie_saludable_75772_600.jpg",
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 15,
            "height": 150,
            "width": 150,
          },
          null,
        ]
      }
    />
    <Text
      style={
        Object {
          "color": "white",
          "marginTop": 10,
        }
      }
    >
      Lemon Pie
    </Text>
  </View>
</View>
`;

exports[`Recipe RecipeList Component Render 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "id": 1,
        "img": "https://t2.rg.ltmcdn.com/es/posts/2/7/7/lemon_pie_saludable_75772_600.jpg",
        "ingredients": Array [
          Object {
            "amount": "1",
            "id": 1,
            "name": "egg white",
          },
          Object {
            "amount": "70 grams",
            "id": 2,
            "name": "wholemeal flour",
          },
          Object {
            "amount": "30 grams",
            "id": 3,
            "name": "oatmeal",
          },
          Object {
            "amount": "1 tablespoon",
            "id": 4,
            "name": "sugar",
          },
          Object {
            "amount": "1",
            "id": 5,
            "name": "yolk",
          },
          Object {
            "amount": "200 milliliters",
            "id": 6,
            "name": "skim milk or unsweetened vegetable milk",
          },
          Object {
            "amount": "1/2 juice",
            "id": 7,
            "name": "lemon juice and zest",
          },
        ],
        "like": 0,
        "servings": "6",
        "title": "Lemon Pie",
      },
      Object {
        "id": 2,
        "img": "https://sibeti.com/wp-content/uploads/2021/09/espagueti-verde-con-chile-poblano-e1631910735274.jpg",
        "ingredients": Array [
          Object {
            "amount": "1",
            "id": 1,
            "name": "Cups of tomato puree 1/2",
          },
          Object {
            "amount": "500 grams",
            "id": 2,
            "name": "Cup of water",
          },
          Object {
            "amount": "1",
            "id": 3,
            "name": "Tomato Cubes with Chicken",
          },
          Object {
            "amount": "1",
            "id": 4,
            "name": "Can of Media Crema",
          },
          Object {
            "amount": "200 grams",
            "id": 5,
            "name": "spaghetti pack",
          },
          Object {
            "amount": "2",
            "id": 6,
            "name": "Fresh parsley",
          },
        ],
        "like": 0,
        "servings": "6",
        "title": "spaghetti",
      },
      Object {
        "id": 3,
        "img": "https://www.cookingclassy.com/wp-content/uploads/2018/04/teriyaki-chicken-15.jpg",
        "ingredients": Array [
          Object {
            "amount": "6",
            "id": 1,
            "name": "boneless chicken thighs",
          },
          Object {
            "amount": "2",
            "id": 2,
            "name": "garlic cloves",
          },
          Object {
            "amount": "50 grams",
            "id": 3,
            "name": "runny honey",
          },
          Object {
            "amount": "1 tablespoon",
            "id": 4,
            "name": "rice wine vinegar",
          },
          Object {
            "amount": "1 tablespoon",
            "id": 5,
            "name": "sesame seeds",
          },
          Object {
            "amount": "4",
            "id": 6,
            "name": "sping onions",
          },
          Object {
            "amount": "2 tablespoon",
            "id": 7,
            "name": "toasted sesame oil",
          },
        ],
        "like": 0,
        "servings": "6",
        "title": "Teriyaki chicken",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={true}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "flexGrow": 0,
      "marginLeft": "5%",
      "marginRight": "5%",
      "padding": 10,
      "width": "90%",
    }
  }
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginRight": 10,
            "opacity": 1,
          }
        }
      >
        <View>
          <Image
            source={
              Object {
                "uri": "https://t2.rg.ltmcdn.com/es/posts/2/7/7/lemon_pie_saludable_75772_600.jpg",
              }
            }
            style={
              Array [
                Object {
                  "borderRadius": 15,
                  "height": 150,
                  "width": 150,
                },
                null,
              ]
            }
          />
          <Text
            style={
              Object {
                "color": "white",
                "marginTop": 10,
              }
            }
          >
            Lemon Pie
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginRight": 10,
            "opacity": 1,
          }
        }
      >
        <View>
          <Image
            source={
              Object {
                "uri": "https://sibeti.com/wp-content/uploads/2021/09/espagueti-verde-con-chile-poblano-e1631910735274.jpg",
              }
            }
            style={
              Array [
                Object {
                  "borderRadius": 15,
                  "height": 150,
                  "width": 150,
                },
                null,
              ]
            }
          />
          <Text
            style={
              Object {
                "color": "white",
                "marginTop": 10,
              }
            }
          >
            spaghetti
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginRight": 10,
            "opacity": 1,
          }
        }
      >
        <View>
          <Image
            source={
              Object {
                "uri": "https://www.cookingclassy.com/wp-content/uploads/2018/04/teriyaki-chicken-15.jpg",
              }
            }
            style={
              Array [
                Object {
                  "borderRadius": 15,
                  "height": 150,
                  "width": 150,
                },
                null,
              ]
            }
          />
          <Text
            style={
              Object {
                "color": "white",
                "marginTop": 10,
              }
            }
          >
            Teriyaki chicken
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
